File: legacy/legacy-apis/Psi.Models.ClientConfigurationModels/MobileConfigurations/Security/QuickAccess/QuickAccessPin.cs
namespace Psi.Data.Models.ClientConfigurationModels.MobileConfigurations.Security.QuickAccess
{
    public class QuickAccessPin : SettingsBaseHelper
    {
        public QuickAccessPin(ISettingsBase settingsBase) : base(settingsBase)
        {
        }

        [SettingKey("Mobile.Security.QuickAccess.Pin.Enabled")]
        public bool Enabled
        {
            get { return GetBoolValue(); }
            set { SetValue(value); }
        }

		/// <summary>
		/// Pin type can be numeric or alphanumeric
		/// </summary>
		[SettingKey("Mobile.Security.QuickAccess.Pin.Type")]
        public string Type
        {
            get { return GetValue(); }
            set { SetValue(value); }
        }

        [SettingKey("Mobile.Security.QuickAccess.Pin.Length")]
        public int Length
        {
            get { return GetIntValue(); }
            set { SetValue(value); }
		}

		/// <summary>
		/// If a user has not created a pin since before this date, the user must create a new pin.
		/// </summary>
		[SettingKey("Mobile.Security.QuickAccess.Pin.ResetDate")]
	    public System.DateTime ResetDate {
		    get { return GetDateTimeValue(); }
		    set { SetValue(value); }
	    }

	}
}

File: workspace/legacy-analyzer/output/infrastructure/models/QuickAccess/QuickAccessPin.ts
// Generated imports
import { ResetDate } from '../ResetDate';

export interface QuickAccessPin {
    /** @settingKey Mobile.Security.QuickAccess.Pin.Enabled */
    enabled: boolean;
    /** @settingKey Mobile.Security.QuickAccess.Pin.Type */
    /**
     * /// <summary>
     * /// /// <summary>
     * /// /// Pin type can be numeric or alphanumeric
     * /// /// </summary>
     * /// </summary>
     */
    type: string;
    /** @settingKey Mobile.Security.QuickAccess.Pin.Length */
    length: number;
    /** @settingKey Mobile.Security.QuickAccess.Pin.ResetDate */
    /**
     * /// <summary>
     * /// /// <summary>
     * /// /// If a user has not created a pin since before this date, the user must create a new pin.
     * /// /// </summary>
     * /// </summary>
     */
    system: ResetDate;
}
